### PR TITLE
Recent cases heat map

### PR DESCRIPTION
**Pull request by panerorenn9541**
- Add heatmap dropdown option that shows heat based only on stats over the past 7 days
- Data calculated by summing up the difference in cases between a day and the previous day for the past week
- Accumulate and display all country/county data in heatmap, supporting relative heat

### CODE REVIEW

**Comment by panerorenn9541** [comment]
Screenshot 2020-08-28 at 10 20 50 PM

Showing recent cases

**Comment by panerorenn9541** [comment]
Screenshot 2020-08-28 at 10 21 20 PM

Showing all time cases

File: src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java [hunk 1]
```diff
@@ -75,6 +75,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Gson gson = new Gson();
       String timeReportJson = gson.toJson(toReturn);
       response.getWriter().println(timeReportJson);
+      // Cases in last 7 days for heatmap
+    } else if (lat == 1000.0 && lng == 1000.0) {
```
Comment (huzcn) [inline src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java]: It's better to represent this value as a constant, and give it a meaningful name + add comments: why 1000? what's the magic here?

File: src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java [hunk 2]
```diff
@@ -75,6 +75,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+      List<recentReport> recentReports = new ArrayList<recentReport>();
+      for (LocLatLng gkey : globalTimeReports.keySet()) {
+        if (gkey.location.contains("US")) {
+          continue;
+        }
+        int casesSum = 0;
+        int arrSize = globalTimeReports.get(gkey).size();
+        for (int i = arrSize - 7; i < arrSize; ++i) {
```
Comment (huzcn) [inline src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java]: Use a constant to represent 7

File: src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java [hunk 3]
```diff
@@ -75,6 +75,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+          casesSum += (globalTimeReports.get(gkey).get(i) - globalTimeReports.get(gkey).get(i - 1));
+        }
+        if (casesSum < 0) {
```
Comment (huzcn) [inline src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java:98]: Is this situation possible?

**Review by huzcn — APPROVED** [review]
Thanks for the fix! This can be mentioned in the demo.

File: src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java (hunk 3 above)
Comment (panerorenn9541) [reply src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java:98]: This can happen if a government removes false positive test results

**Review by alsile — APPROVED** [review]
nice change, heat map will make more sense

**Review by kinsaurralde — APPROVED** [review]

**Pull request merged**